src/encoder.rs

```diff
@@ -1869,43 +1869,48 @@ fn encode_partition_topdown(fi: &FrameInvariants, fs: &mut FrameState,
 
 use rayon::prelude::*;
 
-fn encode_tile(fi: &FrameInvariants, fs: &mut FrameState) -> Vec<u8> {
-  let mut w = WriterEncoder::new();
-
-  let fc = if fi.primary_ref_frame == PRIMARY_REF_NONE {
-    CDFContext::new(fi.base_q_idx)
-  } else {
-    match fi.rec_buffer.frames[fi.ref_frames[fi.primary_ref_frame as usize] as usize] {
-      Some(ref rec) => rec.cdfs,
-      None => CDFContext::new(fi.base_q_idx)
-    }
-  };
-
-  let bc = BlockContext::new(fi.w_in_b, fi.h_in_b);
-  // For now, restoration unit size is locked to superblock size.
-  let mut cw = ContextWriter::new(fc, bc);
 
-  // initial coarse ME loop
+#[inline(always)]
+fn build_coarse_pmvs(fi: &FrameInvariants, fs: &FrameState) -> Vec<[Option<MotionVector>; REF_FRAMES]> {
+// initial coarse ME loop
+  assert!(!fi.sequence.use_128x128_superblock);
   let sby_range = 0..fi.sb_height;
   let sbx_range = 0..fi.sb_width;
 
   let sbos = (sby_range).flat_map(|sby| {
       sbx_range.clone().map(move |sbx| SuperBlockOffset { x: sbx, y: sby })
   }).collect::<Vec<SuperBlockOffset>>();
 
-  let frame_pmvs: Vec<[Option<MotionVector>; REF_FRAMES]> = sbos.par_iter().map(|sbo| {
+  sbos.par_iter().map(|sbo| {
```

Comment:
Maybe, removing this assignment empowers to use x2 cores?

Reply:
No, there is not change in that, just that on speed 3 and 5 we do much less between the calls.

Reply:
Oh, I see. So, for larger speed or QPs, time for initial qpel and hpel pmv motion estimation is relatively larger part than time for encoding each SB, this is what you meant?

Reply:
Yes, the mv computation is a substantial part of the encoding process for speed 3 and higher.